### PR TITLE
Split up the monolithic index.md, take two

### PR DESCRIPTION
**Pull request by magicus**
Here's a resurrection of my old code to split up the monolithic index.md.

I have fully recreated the per-chapter files based on the current content of index.md. I have removed the number prefix and instead the order of the files are determined by `src/toc.conf`.

By my quick inspection, the generated index.html looks identical to the one before.

I found some inconsistencies when doing this. For instance most, but not all, chapters end with a 
```
::: {.box}
[To the top](#){.boxheader}
:::
```
footer. A reasonable follow-up is to move these per-chapter footers out of the actual markdown, and inject them when combining the file instead. This means less redundancy, and no more inconsistencies. ;-)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Reviewers
 * [Jesper Wilhelmsson](https://openjdk.java.net/census#jwilhelm) (@JesperIRL - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/guide pull/82/head:pull/82` \
`$ git checkout pull/82`

Update a local copy of the PR: \
`$ git checkout pull/82` \
`$ git pull https://git.openjdk.java.net/guide pull/82/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 82`

View PR using the GUI difftool: \
`$ git pr show -t 82`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/guide/pull/82.diff">https://git.openjdk.java.net/guide/pull/82.diff</a>

</details>

### CODE REVIEW

**Review by JesperIRL — APPROVED** [review]
Thanks for doing this Magnus! Looks good!

File: Makefile [hunk 1]
```diff
@@ -7,14 +7,28 @@ endef
 
 TITLE := The OpenJDK Developers' Guide
 
-ALL_MD := $(wildcard src/*.md)
-ALL_FOOTER := $(patsubst src/%.md, build/support/footers/%.html, $(ALL_MD))
-ALL_RESULT := $(patsubst src/%.md, build/dist/%.html, $(ALL_MD))
+LEGACY_MD := $(wildcard src/*.md)
+LEGACY_FOOTER := $(patsubst src/%.md, build/support/footers/%.html, $(LEGACY_MD))
+LEGACY_RESULT := $(patsubst src/%.md, build/dist/%.html, $(LEGACY_MD))
```
[the remark this answers is not in the document]
Comment (JesperIRL) [reply Makefile:12]: Oh, I replied in the other mail thread :-)  Those files can be removed.

[the remark this answers is not in the document]
Comment (magicus) [reply Makefile:12]: I'll do that first, then.

**Comment by magicus** [comment]
I'll push this change now. We can clean up the rest later, when the legacy files can be fully removed.

**Pull request closed without merging**